### DISCUSSION TITLE
Coreference Resolution with Spacy v3

### DISCUSSION THREAD

**Discussion by morankyle** [explosion/spaCy#8502] [New Features & Project Ideas]
With Neuralcoref currently not compatible with Spacy v3, are there any other packages you would recommend? Perhaps AllenNLP? 

Also is there any rough timeframe on Neuralcoref becoming compatible with Spacy v3?

**Comment by polm** [comment]
As explained in [an issue on neuralcoref](https://github.com/huggingface/neuralcoref/issues/295#issuecomment-859005189), development on it is not continuing. 

We're working on a coref model for v3 right now but don't have a date for it yet, though it shouldn't be too long. 

The [coreferee](https://github.com/msg-systems/coreferee) package recently added to Universe supports v3 and may be worth a look.

**Reply by metalaureate** [reply]
Is it possible to run spaCy 2.x and 3.x simultaneoulsy to have a working coference solution? (short of creating an isolated microservice for it) Or could I test a beta version of the package you are working on?

I could not **coreferee** to install. There appears to be no valid distribution. When I follow the install instructions:

```
python3 -m pip install coreferee
ERROR: Could not find a version that satisfies the requirement coreferee (from versions: none)
ERROR: No matching distribution found for coreferee
```

**Reply by morankyle** [reply]
I was able to use Coreferee successfully with Spacy v3, if you need help with that I could walk you through it, I don't believe it works with Spacy v2 though.... I could be wrong though.

**Reply by metalaureate** [reply]
@morankyle Thank you! How did you install it? I got the error I posted above and felt DOA.

**Reply by morankyle** [reply]
Had to do it without pip as behind firewall but...

1. Cloned the repo (https://github.com/msg-systems/coreferee/tree/master)
2. Navigate to root directory
3. Run (‘python -m coreferee install en’)

After this it worked, something else I ran into is if using with a custom model be sure to have all the needed pipes that Coreferee relies on, currently it will not error out but instead just return nothing or odd output for doc._.coref_chains.print(), etc.

**Reply by metalaureate** [reply]
Thank you.

**Reply by metalaureate** [reply]
I get error on `nlp.add_pipe('coreferee')`
`coreferee.errors.ModelNotSupportedError: en_core_web_trf version 3.1.0
`

**Reply by morankyle** [reply]
Checkout out the config.cfg file in (coreferee/coreferee/lang/en), basically it checks which model your using (meta.json of the Spacy model) and then pairs the Spacy model with one of the Coreferee models, your issue is that currently its only checking for Spacy models that are 3.0.0. So you could edit that file so it allows versions up to 3.0.1

**Reply by metalaureate** [reply]
Wonderful, that worked. Thank you!

**Comment by svlandeg** [comment]
Update: our coref component for v3 is now released in `spacy-experimental`, cf https://github.com/explosion/spaCy/discussions/11585!
